### ISSUE TITLE
Logging additional values with an error object

### ISSUE THREAD

**Issue by parshap** [rvagg/bole#12]
Hi! 👋 

I'd like to log additional context when logging an error object so that I have more information about why the error occurred. I have something like the following:

``` js
var log = require('bole')('mymodule')
var query = 'select * from users'
executeQuery(query, function(err, results) {
  if (err) {
    // Want to log query that caused error
    err.query = query;
    log.error(err);
  }
})
```

In this example, I want the value of `query` to show up in my logs alongside the error object.

I don't currently see a way of doing this. Should it be possible?

**Comment by ckross01** [comment]
you can just log both values

`log.error(err, query)`

**Comment by parshap** [comment]
> you can just log both values
> 
> `log.error(err, query)`

If I understand correctly, `query` will then get used as the log's `message` property. That can work for some cases, but what if I have multiple metadata values that I want to attach to the log? For example: the latency of the call, the server the call was made to, an `errno` that e.g., MongoDB exposes, etc. I know I can format all of these into a single string and use that as the `message`, but I want to be able to filter on these values individually.

**Comment by rvagg** [comment]
Right, I understand the problem you're having here, there are a couple of ways to solve it with what we have already, but there's scope to extend the API to take care of this a bit better.

For now, you could either:

1. Use the second argument to lay out the `message` property that will contain the values that you insert with the subsequent arguments, it simply uses [`util.format()`](https://nodejs.org/api/util.html#util_util_format_format) so you can do something like `log.error(err, 'errno=%d, latency=%d', errno, latency)`. Unfortunately you lose out on the nice JSON formatting of these properties but perhaps it doesn't matter too much in your situation because they are extractable anyway?
2. Send a custom object containing the error, something like `log.error({ error: { name: err.name, message: err.message, code: err.code, stack: err.stack }, errno: errno, latency: latency })`. This would basically do what you want and replicate what bole is doing under the hood with errors anyway (see the `errorToOut()` function in the bole source. This is pretty verbose but it'll work.

I'll leave this open because one of us might come up with a nice way to integrate an efficient and nice API to deal with this kind of situation.

**Comment by rvagg** [comment]
added new functionality from #16 for this

**Comment by parshap** [comment]
@rvagg I’m not seeing anything added to master recently 🤔

**Comment by rvagg** [comment]
done and published, v4.0.0
